src/components/selectionBoxLayer.ts

```diff
@@ -0,0 +1,141 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Component {
+  export class SelectionBoxLayer extends AbstractComponent {
+    private _box: D3.Selection;
+    private _boxArea: D3.Selection;
+    private _boxEdgeT: D3.Selection;
+    private _boxEdgeB: D3.Selection;
+    private _boxEdgeL: D3.Selection;
+    private _boxEdgeR: D3.Selection;
+    private _boxCornerTL: D3.Selection;
+    private _boxCornerTR: D3.Selection;
+    private _boxCornerBL: D3.Selection;
+    private _boxCornerBR: D3.Selection;
+
+    private _boxBounds: Bounds = {
+      topLeft: { x: 0, y: 0 },
+      bottomRight: { x: 0, y: 0 }
+    };
+    private _boxEdgeWidth = 2;
+
+    constructor() {
+      super();
+      this.classed("selection-box", true);
+    }
+
+    protected _setup() {
+      super._setup();
+
+      this._box = this._content.append("g").classed("selection-box", true).remove();
+      this._boxArea = this._box.append("rect").classed("selection-area", true);
+
+      var createLine = () => this._box.append("line").style({
+                               "opacity": 0,
+                               "stroke": "pink",
+                               "stroke-width": this._boxEdgeWidth
+                             });
+      this._boxEdgeT = createLine().classed("selection-edge-tb", true);
+      this._boxEdgeB = createLine().classed("selection-edge-tb", true);
+      this._boxEdgeL = createLine().classed("selection-edge-lr", true);
+      this._boxEdgeR = createLine().classed("selection-edge-lr", true);
+
+      var createCorner = () => this._box.append("circle")
+                                   .attr("r", this._boxEdgeWidth / 2)
+                                   .style({
+                                     "opacity": 0,
+                                     "fill": "pink"
```

Comment:
`"pink"` is shorter, and we can be cheeky about it.

"What's going on here?"
"Invisible pink ____"